src/backend/cdb/cdbutil.c

```diff
@@ -952,6 +964,75 @@ cdbcomponent_activeQEsExist(void)
 	return !cdb_component_dbs ? false : cdb_component_dbs->numActiveQEs > 0;
 }
 
+bool
+cdb_checkWarehouseName(char *new_name)
+{
+	if (Gp_role == GP_ROLE_EXECUTE)
+		return true;
+	if (new_name == NULL || strcmp(new_name, "default") == 0)
+		return true;
+
+	Oid warehouseid = GetGpWarehouseOid(new_name, true);
+
+	if (!OidIsValid(warehouseid))
+		GUC_check_errmsg("warehouse %s does not exist", new_name);
+
+	return true;
+}
+
+void
+cdb_assignCurrentWarehouse(char *old_name, char *new_name)
```

Comment:
We don't need to  place cdb_assignCurrentWarehouse in CBDB.

Reply:
when we set the GUC value, we can not call this function from the extension.